File: 151. Reverse Words in a String/code.cpp
class Solution {
public:
    string reverseWords(string s) {
        string res="";
        int size=s.size();
        string g="";
        for(int i=0;i<size;i++)
        {
            if(s[i]!=' ')
            {
                g=g+s[i];
            }else{
                
                if(g.size()>0)
                {
                    if(res.size()>0)
                    {
                        res=' '+res;
                        res=g+res;
                        g="";
                    }
                    else{
                        res=g+res;
                        g="";
                    }
                }
            }
        }
        if(g.size()>0&&res.size()>0)
        {
            res=' '+res;
            res=g+res;
            g="";
        }
        else{
            res=g+res;
            g="";
        }
        return res;
    }
};

File: 151. Reverse Words in a String/code.java
class Solution {
    public String reverseWords(String s) {
        int size=s.length();
        String g="",res="";
        for(int i=0;i<size;i++)
        {
            if(s.charAt(i)!=' ')
            {
                g+=s.charAt(i);
            }
            else{
                if(g.length()>0)
                {
                    if(res.length()>0)
                    {
                        res=' '+res;
                        res=g+res;
                    }
                    else{
                        res=g+res;
                    }
                    g="";
                }
            }
        }
        if(g.length()>0)
        {
            if(res.length()>0)
            {
                res=' '+res;
                res=g+res;
            }
            else{
                res=g+res;
            }
            g="";
        }
        return res;
    }
}
